### ISSUE TITLE
windows 10 use bash terminal

### ISSUE THREAD

**Issue by wisheen** [Microsoft/vscode-maven#48]
I use bash (windows subsystem) as default terminal, compile maven project will be error:

mvn -f "d:\workspace\lts-core\pom.xml" compile
wsh@WSH-T303UA:/mnt/d/workspace$ mvn -f "d:\workspace\lts-core\pom.xml" compile
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-readable POM /mnt/d/workspace/d:\workspace\lts-core\pom.xml: /mnt/d/workspace/d:\workspace\lts-core\pom.xml (No such file or directory) @
 @

use this command will be correct:
mvn -f "/mnt/d/workspace/lts-core/pom.xml" compile

Please be compatible with bash.

**Comment by Eskibear** [comment]
Currently the extension judge the terminal type according to its' path (in fact making a guess from it), and format the command. But WSL's filepath format is different, path like `C:/test/pom.xml` works well with **Git Bash for Windows** but not with WSL. We may need to deal with the path format in a more general approach.
 
Anyway, the first step is to distinguish **Git Bash for Windows** and **WSL**, for they have same executable name `bash.exe`.

  
@wisheen How do you specify the integrated terminal path?

Simply like this?
 ```
"terminal.integrated.shell.windows": "bash.exe"
```

OR specify the full path like this?
```
 "terminal.integrated.shell.windows": "C:\\Windows\\System32\\bash.exe"
```

**Comment by AlexanderNull** [comment]
I'm hitting the same problem. I've got: `"C:\\windows\\Sysnative\\bash.exe"` as my shell value.

**Comment by AlexanderNull** [comment]
`c:\` related pathing appears to be generated from `vscode_1.workspace.workspaceFolders` items which have the `uri.path` property set to your os format, not your shell format. executeGoal ends up getting an item with an abosolutePath in your os format as a result which gets passed to runInTerminal. This is my first time looking at the inner workings of a VSC extension so I'm not fully sure what a good fix or avenue of attack is though.

**Comment by Eskibear** [comment]
Here's how vscode detects terminal types:

https://github.com/Microsoft/vscode/blob/9ceaf07eed820ecd1aba6cec302bc0d880ccded1/src/vs/workbench/parts/terminal/electron-browser/terminalService.ts#L209

**Comment by Eskibear** [comment]
shipped in v0.9.0, have a try.
